### DISCUSSION TITLE
Apkeditor GUI

### DISCUSSION THREAD

**Discussion by shmu1i** [REAndroid/APKEditor#141] [Ideas]
I think it would be great to have a GUI for apkeditor similar to [Apk Easy Tool](https://xdaforums.com/t/closed-discontinued-windows-apk-easy-tool-v1-60-2022-06-23.3333960/). I have started working on a python project for this and will share it once it is semi useful. @REAndroid let me know if you would want this part of the main project or a separate project.
![image](https://github.com/user-attachments/assets/e73a1120-6155-4bb9-babb-9def981e6edd)

**Comment by REAndroid** [comment]
Nice idea, but I have a lot of core features in mind to implement from dex manipulation to native library editing. Implementing GUI is a huge burden to handle. 
I encourage and support if you have a plan to implement it yourself. I recommend to use java GUI instead of other programming languages or scripts, it will allow you to use full potential of APKEditor.

Thank you!

**Comment by VictorH028** [comment]
If you are still interested, I sign up for the development of the graphical interface

**Comment by REAndroid** [comment]
@VictorH028 I will try to help as much as i can

**Reply by yonggamer** [reply]
i made a working gui with ai
[gui.zip](https://github.com/user-attachments/files/32485483/gui.zip)

**Reply by REAndroid** [reply]
Would be convenient to comment if you commit the source on your repo.

**Reply by yonggamer** [reply]
https://github.com/REAndroid/APKEditor/pull/240

**Reply by VictorH028** [reply]
It's great that you're interested, but do you know how to program without AI?

**Reply by REAndroid** [reply]
> #240


This is impressive for start.

This should be a separate project with dedicated maintainer, personally I don't have a capacity to handle on top of current projects.
I suggest to make separate your repo or fork it with new branch. I will make sure future commits will not disturb your branch.

Just make a quick look at your pull [MainFrame.runCommand](https://github.com/yonggamer/APKEditor/blob/master/gui/src/com/apkeditor/gui/MainFrame.java#L152). Since you are developing with java, you have to use all the benefits. No need of jar execution or cli command parsing.

Example to decompile:

```java 
        DecompileOptions options = new DecompileOptions();
        options.inputFile = new File("path"); // input file obtained from gui
        options.type = TYPE_XML;
        options.outputFile = new File("path"); // output file obtained from gui

        // override to intercept logs
        Decompiler decompiler = new Decompiler(options){
            @Override
            public void logMessage(String msg) {
                JTextArea area = null; // set your log JTextArea
                area.append("\n" + msg);
            }
        };
        decompiler.setEnableLog(true);
        decompiler.runCommand();
```

**Reply by yonggamer** [reply]
i have now this https://github.com/yonggamer/gui-for-APKEditor

**Reply by VictorH028** [reply]
Use the API. Not (    System )
